### PR TITLE
Updating AvatarWithBadge to BadgeWrapper

### LINKED ISSUE #17706: Update component: BadgeWrapper
### **Description**

Create `BadgeWrapper` component based on audit and insight report 

### **Technical Details**
- TBC

### **Acceptance Criteria**
- TBC
 
### **References**

[FigJam](https://www.figma.com/file/pQa2vI7r08pHiqsAPTWdQ8/Avatar%2C-Token%2C-Network%2C-Badge-Audit?node-id=0%3A1&t=xNhlymemDFsX58Sc-1)

### PR DESCRIPTION
**Pull request by georgewrmarshall**
## Explanation

Updating `AvatarWithBadge` to be called `BadgeWrapper` and updating component api according to insight report. Also introduces the first enums to the component library.

* Fixes #17706 
* [Insight report FigJam](https://www.figma.com/file/pQa2vI7r08pHiqsAPTWdQ8/Avatar%2C-Token%2C-Network%2C-Icon%2C-Badge-Audit?node-id=602%3A10&t=jOYT7v3ZGEeeyIfR-1)

Screenshot 2023-02-21 at 9 50 20 PM

### Before

https://user-images.githubusercontent.com/8112138/220523096-b229db33-4238-4794-9a9b-98ae980e1cbc.mov

### After

https://user-images.githubusercontent.com/8112138/220523338-c2f9a6aa-f320-401a-a479-9863149338cb.mov

No instances of `AvatarWithBadge` in code base

Screenshot 2023-02-21 at 9 28 56 PM

## Manual Testing Steps

- Go to the latest build of storybook in this PR
- Search for `BadgeWrapper` in the search bar in the side panel
- Check the storybook controls
- Check the stories render properly
- Check the documentation

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/badge-wrapper/README.mdx [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+      name="Arbitrum One"
+      src="./images/avax-token.png"
+      borderColor={BorderColor.borderMuted}
+    />
+  }
+  positionObj={{ top: 4, right: -8 }}
+>
+  <AvatarAccount
+    address="0x5CfE73b6021E818B776b421B1c4Db2474086a7e1"
+    size={Size.SM}
```
Comment (NidhiKJha) [inline ui/components/component-library/badge-wrapper/README.mdx:313]: Can we make this default size to `size={Size.MD}` to keep it in sync with the other stories?

File: ui/components/component-library/badge-wrapper/badge-wrapper.stories.js [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+          name="Arbitrum One"
+          src="./images/avax-token.png"
+          borderColor={BorderColor.borderMuted}
+        />
+      }
+      positionObj={{ top: 4, right: -8 }}
+    >
+      <AvatarAccount
+        address="0x5CfE73b6021E818B776b421B1c4Db2474086a7e1"
+        size={Size.SM}
```
Comment (NidhiKJha) [inline ui/components/component-library/badge-wrapper/badge-wrapper.stories.js:295]: Can we make this default size to `size={Size.MD}` to keep it in sync with the other stories?

**Review by NidhiKJha — COMMENTED** [review]
Looking really great 🤩 . Just a small nit

File: ui/components/component-library/badge-wrapper/README.mdx (hunk 1 above)
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/README.mdx:298]: Yes! Fixed here https://github.com/MetaMask/metamask-extension/pull/17851/commits/d01346d0cff09d59c6741d4c6b2e3e0bc7f36bd8

File: ui/components/component-library/badge-wrapper/badge-wrapper.stories.js (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/badge-wrapper.stories.js]: Yes! Fixed here https://github.com/MetaMask/metamask-extension/pull/17851/commits/d01346d0cff09d59c6741d4c6b2e3e0bc7f36bd8

**Comment by Sylvia24** [comment]
We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

**Review by darkwing — DISMISSED** [review]

**Review by garrettbear — DISMISSED** [review]

File: ui/components/component-library/badge-wrapper/README.mdx [hunk 3]
```diff
@@ -0,0 +1,381 @@
...
+### Children
+
+Use the `children` prop to define the element to be wrapped by the `BadgeWrapper`. This element will be what the badge is positioned on top of.
+
+<Canvas>
+  <Story id="components-componentlibrary-badgewrapper--children" />
+</Canvas>
+
+```jsx
+import { Color Size } from '../../../helpers/constants/design-system';
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/README.mdx]: 🦅 👀 🙇

File: ui/components/component-library/badge-wrapper/badge-wrapper.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
+import React from 'react';
```
Comment (brad-decker) [inline ui/components/component-library/badge-wrapper/badge-wrapper.js:1]: Should we convert these new files to typescript now?

**Review by brad-decker — DISMISSED** [review]

File: ui/components/component-library/badge-wrapper/badge-wrapper.js (hunk 4 above)
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/badge-wrapper.js:1]: Will this work if it's using the `Box` component as a base?

File: ui/components/component-library/badge-wrapper/index.ts [hunk 5]
```diff
@@ -0,0 +1,5 @@
+export { BadgeWrapper } from './badge-wrapper';
+export {
+  BadgeWrapperPosition,
+  BadgeWrapperAnchorElementShape,
```
[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/index.ts:4]: Yes probably! Good call out

[the remark this answers is not in the document]
Comment (georgewrmarshall) [reply ui/components/component-library/badge-wrapper/index.ts:4]: Updated.

**Comment by helpme67** [comment]
Check

**Comment by georgewrmarshall** [comment]
@brad-decker now that https://github.com/MetaMask/metamask-extension/pull/18009 has been merge this component has been updated to TypeScript

**Comment by georgewrmarshall** [comment]
Not sure why this is failing test coverage and can't see how any of the file here affect it I would suggestion reviewing and merging regardless??

Screenshot 2023-03-08 at 12 15 13 PM

**Review by darkwing — APPROVED** [review]

**Review by garrettbear — APPROVED** [review]

**Pull request merged**